### PR TITLE
ci: work around a bunch of unreliable test failures

### PR DESCRIPTION
**Pull request by emmettbutler**
This change adds workarounds for four unreliable test failures recently observed on the main branch.

* https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/56696/workflows/2a0f2e96-d1fc-4d40-8b5b-c414adfa40db/jobs/3599020 - loosens assertion to accept multiple calls, which have been observed
* https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/56703/workflows/de876b66-0a76-405e-aaba-efb28c79701b/jobs/3599380 - marks as flaky, because the failure mode appears to be incompatible with the test's intent
* https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/56727/workflows/a7ef12fa-bd08-4e0b-b126-cc6115051d27/jobs/3600333 - disables the pytest coverage plugin, which is implicated in the traceback
* https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/56760/workflows/7d0322a2-30bf-4afa-b501-b4ba3eb44b84/jobs/3601489 - makes `wait_for_num_traces` in snapshot tests permissive of extra traces, an approach similar to the one taken for other snapshot code paths

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by sanchda — APPROVED** [review]

File: riotfile.py [hunk 1]
```diff
@@ -1552,7 +1552,7 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
         ),
         Venv(
             name="pytest",
-            command="pytest --no-ddtrace {cmdargs} tests/contrib/pytest/",
+            command="pytest --no-ddtrace --no-cov {cmdargs} tests/contrib/pytest/",
```
Comment (erikayasuda) [inline riotfile.py:1555]: Is this related to the incompatibility between ITR and pytest coverage?

File: tests/tracer/runtime/test_container.py [hunk 2]
```diff
@@ -310,7 +310,7 @@ def test_get_container_info(file_contents, container_id, node_inode):
             else:
                 assert isinstance(info.node_inode, int)
 
-        mock_open.assert_called_once_with("/proc/self/cgroup", mode="r")
+        mock_open.assert_called_with("/proc/self/cgroup", mode="r")
```
Comment (erikayasuda) [inline tests/tracer/runtime/test_container.py:313]: What causes this to get called more than once?

**Review by erikayasuda — APPROVED** [review]
LGTM, just have a couple of questions for my own knowledge gaps

File: tests/tracer/runtime/test_container.py (hunk 2 above)
Comment (emmettbutler) [reply tests/tracer/runtime/test_container.py:313]: Don't know. This change is only about making existing tests check for real behavior in cases where the real behavior is still reasonably aligned with the intended functionality being tested. In this case, calling `open()` more than once is not "correct", but doing so also doesn't undermine the functionality of `get_container_info` in a way that would make the test useless.

File: riotfile.py (hunk 1 above)
Comment (emmettbutler) [reply riotfile.py:1555]: It might be, but I don't know.

**Pull request merged**